R/utils.R

```diff
@@ -1154,3 +1154,21 @@ longest_element <- function(x) {
   else
     ""
 }
+
+# Apply rlang::as_quosure across a list or vector, but explicitly pass env
+# and expr to as_quosure
+dots_as_quosures <- function(x) {
+  if (!inherits(x, "lazy_dots")) {
+    stop("Expected lazy dots")
+  }
+  lapply(x, function(x) rlang::as_quosure(x$expr, x$env))
```

Comment:
As is done here https://github.com/r-lib/rlang/blob/64df8e3f/R/compat-lazyeval.R#L51

```suggestion
  lapply(x, function(x) rlang::new_quosure(x$expr, x$env))
```